pcs_test/tier0/lib/test_validate.py

```diff
@@ -1748,6 +1752,37 @@ def test_reports_about_invalid_interval(self):
         )
 
 
+class ValueTimeIntervalOrDuration(TestCase):
+    def get_runner(stdout="", stderr="", returncode=0, env_vars=None):
+        runner = mock.MagicMock(spec_set=CommandRunner)
+        runner.run.return_value = (stdout, stderr, returncode)
+        runner.env_vars = env_vars if env_vars else {}
+        return runner
+
+    def test_no_reports_for_valid_time_interval(self):
+        for interval in ["0", "1s", "2sec", "3m", "4min", "5h", "PT1H2M3S", "6hr"]:
+            with self.subTest(value=interval):
+                assert_report_item_list_equal(
+                    validate.ValueTimeIntervalOrDuration("a", runner=self.get_runner()).validate({"a": interval}),
+                    [],
+                )
+
+    def test_reports_about_invalid_interval(self):
+        assert_report_item_list_equal(
+            validate.ValueTimeIntervalOrDuration("a", runner=self.get_runner()).validate({"a": "invalid_value"}),
+            [
+                fixture.error(
+                    reports.codes.INVALID_OPTION_VALUE,
+                    option_name="a",
+                    option_value="invalid_value",
+                    allowed_values="time interval (e.g. 1, 2s, 3m, 4h, PT1H2M3S, ...)",
+                    cannot_be_empty=False,
+                    forbidden_characters=None,
+                ),
+            ],
+        )
```

Comment:
Here is missing coverage for iso duration values starting with 'P'. It would be enough to re-work it as a subTest with two values ['invalid_value', 'P'] and calling [assert_called_once_with](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once_with) and [assert_not_called](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_not_called) on runner mock. Also runner mock needs to be set to return a non-zero value.